ApiDoctor.Validation/DocFile.cs

```diff
@@ -694,93 +694,268 @@ protected bool ParseMarkdownBlocks(IssueLogger issues)
+        {
+            var mergedHeaders = new List<DocumentHeader>();
+            foreach (var header in headers) 
+            {
+                if (header is ExpectedDocumentHeader expectedHeader)
+                {
+                    mergedHeaders.Add(expectedHeader);
+                }
+                else if (header is ConditionalDocumentHeader conditionalHeader)
                 {
-                    issues.Error(ValidationErrorCode.RequiredDocumentHeaderMissing, $"A required document header is missing from the document: {expectedHeaders[i].Title}");
+                    mergedHeaders.AddRange(FlattenDocumentHeaderHierarchy(conditionalHeader.Arguments));
                 }
             }
+            return mergedHeaders;
         }
 
-        private void AddHeaderToHierarchy(Stack<Config.DocumentHeader> headerStack, Block block)
+        /// <summary>
+        /// Validates a conditional document header against the found headers.
+        /// </summary>
+        /// <param name="expectedHeaders">The list of expected headers.</param>
+        /// <param name="foundHeaders">The list of found headers.</param>
+        /// <param name="expectedIndex">Index of the expected header being validated.</param>
+        /// <param name="foundIndex">Index of the found header being compared.</param>
+        /// <returns>The validation result.</returns>
+        private DocumentHeaderValidationResult ValidateConditionalDocumentHeader(List<object> expectedHeaders, IReadOnlyList<DocumentHeader> foundHeaders, int expectedIndex, int foundIndex)
+        {
+            var validationResult = DocumentHeaderValidationResult.None;
+            var expectedConditionalHeader = expectedHeaders[expectedIndex] as ConditionalDocumentHeader;
+            if (expectedConditionalHeader.Operator == ConditionalOperator.OR)
+            {
+                foreach (var header in expectedConditionalHeader.Arguments)
+                {
+                    // Replace conditional header with this argument for validation
+                    expectedHeaders[expectedIndex] = header;
+                    validationResult = ValidateDocumentHeader(expectedHeaders, foundHeaders, expectedIndex, foundIndex);
+
+                    // If header has been found, stop looking
+                    if (validationResult != DocumentHeaderValidationResult.RequiredDocumentHeaderMissing &&
+                        validationResult != DocumentHeaderValidationResult.OptionalDocumentHeaderMissing)
+                    {
+                        break;
+                    }
+                }
+            }
+            else if (expectedConditionalHeader.Operator == ConditionalOperator.AND)
+            {
+                expectedHeaders[expectedIndex] = expectedConditionalHeader.Arguments.First();
+                expectedHeaders.InsertRange(expectedIndex + 1, expectedConditionalHeader.Arguments.Skip(1));
```

Comment:
we should probably check for the collection size/presence of elements before we call first/skip on the arguments.